apps/src/util/HttpClient.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  AUTHENTICITY_TOKEN_HEADER,
+  getAuthenticityToken,
+} from './AuthenticityTokenStore';
+
+/**
+ * Get a JSON response from the given endpoint and
+ * return it as the specified type. Can also perform
+ * response validation if provided a validator function.
+ */
+export async function getJson<ResponseType>(
+  endpoint: string,
+  init?: RequestInit,
+  validator?: ResponseValidator<ResponseType>
+): Promise<GetResponse<ResponseType>> {
+  const response = await fetch(endpoint, init);
+  if (!response.ok) {
+    throw new Error(response.status + ' ' + response.statusText);
+  }
+
+  const json = await response.json();
+  let value = json;
+
+  if (validator) {
+    value = validator(json);
+  }
+
+  return {
+    value,
+    response,
+  };
+}
+
+/**
+ * POST to the given endpoint. Adds the Rails authenticity
+ * token if useAuthenticityToken is true.
+ */
+export async function post(
+  endpoint: string,
+  body: string,
+  useAuthenticityToken = false,
```

Comment:
Curious: does this need a type or can it infer it because of the default value?

Reply:
Yep, the linter will in fact throw an error if I type this as `boolean` since it can be inferred (not the case if this were a different type though, like `boolean | number` for example)